src/nodejs/unit-http/http.js

```diff
@@ -13,14 +13,7 @@ const {
 } = require('./http_server');
 
 function createServer (options, requestHandler) {
-    if (typeof options === 'function') {
-        requestHandler = options;
-        options = {};
-    } else {
-        stderr.write('createServer was called with an options object. Unit ignores this for the time being.');
-    }
-
-    return new Server(requestHandler);
+    return new Server(options, requestHandler);
 }
 
 const http = require("http")
```

Comment:
Sorry, but I'm finding these commits very confusing...

It seems like this should just mostly replace the first commit, except that you want the options as part of the return and setting stderr.

From my point of view it looks like

- Setting stderr should be in its own commit.
- [Take options as well as requestListener](https://github.com/nginx/unit/pull/1091/commits/920be1168ddaf073240009cbf596b3f7d9647b31) should simply add the options parameter, and shouldn't it also return it? as you do in the latter patch...
- [Move options handling to unit's Server constructor](https://github.com/nginx/unit/pull/1091/commits/52550b18897802b5820bec809902513195afe4af) should then just do the meat of what the first patch did...

Reply:
I squashed all of these into a single commit as it's a single unit of work that relates to that. I don't see the need to separate the individual parts of this into their own commits per the feedback. My thinking is:

- commit for the actual functionality
- commit for the test
- commit for the docs